File: ios/RNGiphyVideoManager.m
#import <React/RCTBridgeModule.h>
#import "React/RCTEventEmitter.h"

@interface RCT_EXTERN_REMAP_MODULE(GiphyReactNativeVideoManager, RNGiphyVideoManager, RCTEventEmitter)

RCT_EXTERN_METHOD(muteAll)

RCT_EXTERN_METHOD(pauseAll)

RCT_EXTERN_METHOD(resume)

@end

File: ios/RNGiphyVideoManager.swift
import GiphyUISDK

@objc(RNGiphyVideoManager)
class RNGiphyVideoManager: NSObject {
  @objc
  public static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc
  func muteAll() -> Void {
    DispatchQueue.main.async {
      if SharedGPHVideoPlayer.initialized {
        SharedGPHVideoPlayer.shared.mute(true)
      }
    }
  }

  @objc
  func pauseAll() -> Void {
    DispatchQueue.main.async {
      if SharedGPHVideoPlayer.initialized {
        SharedGPHVideoPlayer.shared.pause()
      }
    }
  }

  @objc
  func resume() -> Void {
    DispatchQueue.main.async {
      if SharedGPHVideoPlayer.initialized {
        SharedGPHVideoPlayer.shared.resume()
      }
    }
  }
}
